persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAO.kt

```diff
@@ -42,7 +42,7 @@ data class MessageEntity(
     }
 
     enum class DownloadStatus {
-        NOT_DOWNLOADED, IN_PROGRESS, DOWNLOADED, FAILED
+        NOT_DOWNLOADED, IN_PROGRESS, DOWNLOADED, SAVED_EXTERNALLY, FAILED
     }
```

Comment:
```suggestion
    enum class DownloadStatus {
        /**
         * There was no attempt done to fetch the asset's data from remote (server) storage.
         */
        NOT_DOWNLOADED,

        /**
         * The asset is currently being downloaded and will be saved either internally or externally.
         * @see SAVED_INTERNALLY
         * @see SAVED_EXTERNALLY
         */
        IN_PROGRESS,

        /**
         * The asset was downloaded and saved in an internal storage, that should be only readable by this Kalium client.
         * @see SAVED_EXTERNALLY
         */
        SAVED_INTERNALLY,

        /**
         * The asset was downloaded and saved in an external storage, that is readable by other software on 
         * the machine that this Kalium client is currently running on.
         *
         * _.e.g_: Asset was saved in Downloads, Desktop or other user-chosen directory.
         */
        SAVED_EXTERNALLY,

        /**
         * The last attempt at fetching and saving this asset's data failed.
         */
        FAILED
    }
```


`status == NOT_DOWNLOADED` is different from `status != DOWNLOADED`.

To don't mix things up and make a more relatable naming, I propose renaming `DOWNLOADED` to `SAVED_INTERNALLY`.

Also, does this documentation make sense?